server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -872,9 +872,11 @@ private static Mapper getLeafMapper(final ParseContext context,
         // if a leaf field is not mapped, and is defined as a runtime field, then we
         // don't create a dynamic mapping for it and don't index it.
         String fieldPath = context.path().pathAsText(fieldName);
-        RuntimeField runtimeField = context.root().getRuntimeField(fieldPath);
-        if (runtimeField != null) {
-            return new NoOpFieldMapper(subfields[subfields.length - 1], runtimeField.asMappedFieldType().name());
+        MappedFieldType fieldType = context.mappingLookup().getFieldType(fieldPath);
```

Comment:
Will this work with flattened fields? I guess they should already be picked up by the parent.

Reply:
I **think** that flattened have a mapper that would be found above?

Reply:
I also think that flattened fields will be found earlier. Would you be up for adding a test for this, as it seems quite subtle?

Reply:
I opened https://github.com/elastic/elasticsearch/pull/74112

Reply:
Thanks for the extra test !